File: trainer/task.py
#! coding: utf-8
import sys
import json
from random import randint, seed

# Загружаем информацию
data = json.loads(sys.argv[1].replace('\\', '').replace('\'', ''))

# Тут из параметров убирается косая черта, так как в Windows двойные
# кавычки требуют экранирования, а в Linux нет. А еще кавычки в Windows
# убираются, а в Linux нет.

mode = data['mode']
config = data['config']
# print(config)
if mode == 'generate':

    # Получем зерно для генератора случайных чисел

    seed(data['seed'])
    maxval1 = int(config["maxval1"])
    maxval2 = int(config["maxval2"])
    gamemode = config["mode"]
    a = randint(0, maxval1)
    b = randint(0, maxval2)
    if gamemode == 'plus':
        question = str(a) + " + " + str(b) + " = ?"
        answer = str(a + b)
    else:
        question = str(a) + " - " + str(b) + " = ?"
        answer = str(a - b)
    print(json.dumps({'question': question, 'answer': answer}))

elif mode == 'check':

    user_answer = data['user_answer']
    answer = data['answer']

    if user_answer == answer:
        print('true')
    else:
        print('false')

File: trainer/task.cpp
#include <iostream>
#include <string>

#include "json.hpp"
using json = nlohmann::json;
using namespace std;

// Все сложно с типами, но думаю разберетесь
// Доки по nlohmann::json где-то есть, но в принципе все понятно:

int main(int argc, char* argv[]){
    // Распарсить строку в объект
    auto data = json::parse(argv[1]);
    // data["mode"].get<string>() - отдай по ключу mode значение типа string
    if (data["mode"].get<string>() == "generate"){
        int maxval = stoi(data["config"]["maxval1"].get<string>());
        srand(stoi(data["seed"].get<string>()));
        int a = rand() % maxval;
        int b = rand() % maxval;
        string ans = to_string(a) + " + " + to_string(b) + " = ?";
        json result;
        result["question"] = to_string(a) + " + " + to_string(b) + " = ?";
        result["answer"] = to_string(a+b);
        cout << result.dump();
    }  else {
        int user_answer = stoi(data["user_answer"].get<string>());
        int answer = stoi(data["answer"].get<string>());
        user_answer == answer ? cout <<  "true" : cout <<  "false";
        cout << endl;
    }
}